### PR TITLE
fix: Guard against incomplete related terms

### LINKED ISSUE #143: Error: "Cannot read property 'id' of undefined"
If I send the query underneath to the Network of Terms...

https://termennetwerk-api.netwerkdigitaalerfgoed.nl/playground

```graphql

query Terms {
  terms(
    sources: [
      "https://data.cultureelerfgoed.nl/PoolParty/sparql/term/id/cht"
    ]
    query: "kas"
  ) {
    source {
      uri
      name
      creators {
        uri
        name
        alternateName
      }
    }
    result {
      __typename
      ... on Terms {
        terms {
          uri
          prefLabel
          altLabel
          hiddenLabel
          scopeNote
        }
      }
      ... on Error {
        message
      }
    }
  }
}
```

...then an error is returned: `Cannot read property 'id' of undefined`.

### PR DESCRIPTION
**Pull request by ddeboer**
Fix #143.

This is a result of #36.

### CODE REVIEW

File: src/services/terms.ts [hunk 1]
```diff
@@ -90,16 +90,26 @@ export class TermsTransformer {
...
     });
   }
+
+  /**
+   * Map related IRIs to their related terms, making sure to only accept complete related terms.
+   *
+   * Related terms can be incomplete because of the SPARQL query limit (see
+   * https://github.com/netwerk-digitaal-erfgoed/network-of-terms-api/issues/36).
+   */
+  private mapRelatedTerms = (acc: RelatedTerm[], iri: RDF.Term) => {
```
Comment (sdevalk) [inline src/services/terms.ts:106]: What does 'acc' mean?

**Review by sdevalk — COMMENTED** [review]
Nice!

File: src/services/terms.ts (hunk 1 above)
Comment (ddeboer) [reply src/services/terms.ts:106]: Good question! I’m used to call this argument `acc`, short for `accumulator` (from doing functional programming). 

As it turns out, [even JS calls it `accumulator`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/reduce). Would renaming this to `accumulator` help? Or is `terms`/`relatedTerms` clearer?

Comment (ddeboer) [reply src/services/terms.ts]: If you insist. 😁 Moved the reduce into the function.

Comment (sdevalk) [reply src/services/terms.ts]: 🕺

**Review by sdevalk — APPROVED** [review]
Let's go! :)

**Pull request merged**